src/protocols/secure_channel/SimpleSessionResumptionStorage.cpp

```diff
@@ -0,0 +1,273 @@
+}
+
+CHIP_ERROR SimpleSessionResumptionStorage::SaveIndex(const SessionIndex & index)
+{
+    uint8_t buf[MaxIndexSize()];
+    TLV::TLVWriter writer;
+    writer.Init(buf);
+
+    TLV::TLVType arrayType;
+    ReturnErrorOnFailure(writer.StartContainer(kIndexContentTag, TLV::kTLVType_Array, arrayType));
+
+    for (size_t i = index.mSize; i < index.mSize; ++i)
+    {
+        TLV::TLVType innerType;
+        ReturnErrorOnFailure(writer.StartContainer(TLV::AnonymousTag(), TLV::kTLVType_Structure, innerType));
+        ReturnErrorOnFailure(writer.Put(kFabricIndexTag, index.mNodes[i].GetFabricIndex()));
+        ReturnErrorOnFailure(writer.Put(kPeerNodeIdTag, index.mNodes[i].GetNodeId()));
+        ReturnErrorOnFailure(writer.EndContainer(innerType));
+    }
+
+    ReturnErrorOnFailure(writer.EndContainer(arrayType));
+
+    const auto len = writer.GetLengthWritten();
+    VerifyOrReturnError(CanCastTo<uint16_t>(len), CHIP_ERROR_BUFFER_TOO_SMALL);
+
+    DefaultStorageKeyAllocator keyAlloc;
+    ReturnErrorOnFailure(mStorage->SyncSetKeyValue(keyAlloc.SessionResumptionIndex(), buf, static_cast<uint16_t>(len)));
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR SimpleSessionResumptionStorage::LoadIndex(SessionIndex & index)
+{
+    uint8_t buf[MaxIndexSize()];
+    uint16_t len = static_cast<uint16_t>(MaxStateSize());
+
+    DefaultStorageKeyAllocator keyAlloc;
+    ReturnErrorOnFailure(mStorage->SyncGetKeyValue(keyAlloc.SessionResumptionIndex(), buf, len));
+
+    TLV::ContiguousBufferTLVReader reader;
+    reader.Init(buf, len);
+
+    ReturnErrorOnFailure(reader.Next(TLV::kTLVType_Array, kIndexContentTag));
+    TLV::TLVType arrayType;
+    ReturnErrorOnFailure(reader.EnterContainer(arrayType));
+
+    size_t count = 0;
+    CHIP_ERROR err;
+    while ((err = reader.Next(TLV::kTLVType_Structure, TLV::AnonymousTag())) == CHIP_NO_ERROR)
+    {
+        if (count >= ArraySize(index.mNodes))
+        {
+            return CHIP_ERROR_NO_MEMORY;
+        }
+
+        TLV::TLVType containerType;
+        ReturnErrorOnFailure(reader.EnterContainer(containerType));
+
+        FabricIndex fabricIndex;
+        ReturnErrorOnFailure(reader.Next(kFabricIndexTag));
+        ReturnErrorOnFailure(reader.Get(fabricIndex));
+
+        NodeId peerNodeId;
+        ReturnErrorOnFailure(reader.Next(kPeerNodeIdTag));
+        ReturnErrorOnFailure(reader.Get(peerNodeId));
+
+        index.mNodes[count++] = ScopedNodeId(peerNodeId, fabricIndex);
+
+        ReturnErrorOnFailure(reader.ExitContainer(containerType));
+    }
+
+    if (err != CHIP_END_OF_TLV)
+    {
+        return err;
+    }
+
+    ReturnErrorOnFailure(reader.ExitContainer(arrayType));
+    ReturnErrorOnFailure(reader.VerifyEndOfContainer());
+
+    index.mSize = count;
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR SimpleSessionResumptionStorage::SaveLink(ConstResumptionIdView resumptionId, const ScopedNodeId & node)
+{
+    // Save a link from resumptionId to node, in key: /f/<fabricIndex>/r/<resumptionId>
```

Comment:
If I'm understanding correctly, I believe the current code is storing to this key:

```
// Save a link from resumptionId to node, in key: /r/<resumptionId>
```